### PR TITLE
refactor(components): [color-picker] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by weidehai**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.
@sxzz 
I'm going to refactor color-picker component, and this is a beginning. I will continue to commit for it

### CODE REVIEW

File: packages/components/color-picker/src/index.vue [hunk 1]
```diff
@@ -127,263 +125,219 @@ import {
...
-
-    const { inputId: buttonId, isLabeledByFormItem } = useFormItemInputId(
-      props,
-      {
-        formItemContext: formItem,
-      }
-    )
+})
+const props = defineProps(colorPickerProps)
+const emit = defineEmits(['change', 'active-change', UPDATE_MODEL_EVENT])
```
Comment (sxzz) [inline packages/components/color-picker/src/index.vue:143]:
```suggestion
const emit = defineEmits(['change', 'activeChange', UPDATE_MODEL_EVENT])
```

File: packages/components/color-picker/src/index.vue [hunk 2]
```diff
@@ -127,263 +125,219 @@ import {
...
+// active-change is used to prevent modelValue changes from triggering.
+let shouldActiveChange = true
+// data
+const color = reactive(
+  new Color({
+    enableAlpha: props.showAlpha,
+    format: props.colorFormat || '',
+    value: props.modelValue,
+  })
+) as Color
```
Comment (sxzz) [inline packages/components/color-picker/src/index.vue:166]:
```suggestion
const color = reactive<Color>(
  new Color({
    enableAlpha: props.showAlpha,
    format: props.colorFormat || '',
    value: props.modelValue,
  })
)
```

File: packages/components/color-picker/src/props.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { buildProps, isValidComponentSize } from '@element-plus/utils'
+import type { PropType } from 'vue'
+import type { ComponentSize } from '@element-plus/constants'
+
+export const colorPickerProps = buildProps({
```
Comment (sxzz) [inline packages/components/color-picker/src/props.ts:5]: please merge this file to `index.ts`

File: packages/components/color-picker/src/props.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  modelValue: String,
+  id: String,
+  showAlpha: Boolean,
+  colorFormat: String,
+  disabled: Boolean,
+  size: {
+    type: String as PropType<ComponentSize>,
```
Comment (sxzz) [inline packages/components/color-picker/src/props.ts:12]: use `definePropType` instead

File: packages/components/color-picker/src/props.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import type { PropType } from 'vue'
+import type { ComponentSize } from '@element-plus/constants'
+
+export const colorPickerProps = buildProps({
+  modelValue: String,
+  id: String,
+  showAlpha: Boolean,
+  colorFormat: String,
+  disabled: Boolean,
+  size: {
```
Comment (sxzz) [inline packages/components/color-picker/src/props.ts:11]: Actually, we have `useSizeProp`

**Comment by sxzz** [comment]
I reviewed this PR roughly. Maybe you need to read some refactored component.

**Comment by weidehai** [comment]
> I reviewed this PR roughly. Maybe you need to read some refactored components first. That will help you to understand the rules better.

yes,thank you❤

File: packages/components/color-picker/src/index.vue (hunk 1 above)
Comment (weidehai) [reply packages/components/color-picker/src/index.vue]: resolved🙌

File: packages/components/color-picker/src/index.vue (hunk 2 above)
Comment (weidehai) [reply packages/components/color-picker/src/index.vue]: resolved🙌

File: packages/components/color-picker/src/props.ts (hunk 5 above)
Comment (weidehai) [reply packages/components/color-picker/src/props.ts]: resolved🙌

File: packages/components/color-picker/src/props.ts (hunk 3 above)
Comment (weidehai) [reply packages/components/color-picker/src/props.ts]: acrroding to other component,i create a ts file with same name of .vue component file for props definition

File: packages/components/color-picker/src/index.vue (hunk 2 above)
Comment (weidehai) [reply packages/components/color-picker/src/index.vue]: i think it should use as do type assert,becasue reactive<Color> !== Color

**Review by sxzz — APPROVED** [review]

**Pull request merged**